midend/hsIndexSimplify.cpp

```diff
@@ -76,6 +78,14 @@ IR::Node *HSIndexContretizer::eliminateArrayIndexes(HSIndexFinder &aiFinder,
     IR::IfStatement *newIf = nullptr;
     const auto *typeStack = aiFinder.arrayIndex->left->type->checkedTo<IR::Type_Stack>();
     size_t sz = typeStack->getSize();
+    if ((expansion += (sz - 1)) > maxExpansion) {
+        if (expansion - sz < maxExpansion)
+            error(ErrorType::ERR_OVERLIMIT, "%1%too much expansion from non-const array indexes",
+                  statement->srcInfo);
```

Comment:
It's probably worth adding a test case for this error message

Reply:
Added more testcases